components/feature/accounts-push/src/main/java/mozilla/components/feature/accounts/push/FxaPushSupportFeature.kt

```diff
@@ -48,61 +49,49 @@ internal const val PREF_FXA_SCOPE = "fxa_push_scope"
  * @param accountManager The FxaAccountManager.
  * @param pushFeature The [AutoPushFeature] if that is setup for observing push events.
  * @param crashReporter Instance of `CrashReporting` to record unexpected caught exceptions.
+ * @param coroutineScope The scope in which IO work within the feature should be performed on.
  * @param owner the lifecycle owner for the observer. Defaults to [ProcessLifecycleOwner].
  * @param autoPause whether to stop notifying the observer during onPause lifecycle events.
  * Defaults to false so that observers are always notified.
  */
 class FxaPushSupportFeature(
     private val context: Context,
-    accountManager: FxaAccountManager,
-    pushFeature: AutoPushFeature,
+    private val accountManager: FxaAccountManager,
+    private val pushFeature: AutoPushFeature,
     private val crashReporter: CrashReporting? = null,
-    owner: LifecycleOwner = ProcessLifecycleOwner.get(),
-    autoPause: Boolean = false
+    private val coroutineScope: CoroutineScope = CoroutineScope(Dispatchers.IO),
+    private val owner: LifecycleOwner = ProcessLifecycleOwner.get(),
+    private val autoPause: Boolean = false
 ) {
 
     /**
      * A unique scope for the FxA push subscription that is generated once and stored in SharedPreferences.
      *
-     * This scope is randomly generated and unique to the account on that particular device.
+     * This scope is randomly generated and unique to the app install.
      * (why this uuid? Note it is *not* reset on logout!)
-     * (isn't this racey? Different observers reference it)
-     * (maybe you just want the acct uid?)
      */
-    private val fxaPushScope: String by lazy {
-        val prefs = preference(context)
-
-        // Generate a unique scope if one doesn't exist.
-        val randomUuid = UUID.randomUUID().toString().replace("-", "")
-
-        // Return a scope in the format example: "fxa_push_scope_a62d5f27c9d74af4996d057f0e0e9c38"
-        val scope = PUSH_SCOPE_PREFIX + randomUuid
-
-        if (!prefs.contains(PREF_FXA_SCOPE)) {
-            prefs.edit().putString(PREF_FXA_SCOPE, scope).apply()
-
-            return@lazy scope
-        }
+    private val pushScope = PushScopeProperty(context, coroutineScope)
 
-        // The default string is non-null, so we can safely cast.
-        prefs.getString(PREF_FXA_SCOPE, scope) as String
-    }
-
-    init {
-        val autoPushObserver = AutoPushObserver(accountManager, pushFeature, fxaPushScope)
+    /**
+     * Initialize the support feature to launch the appropriate observers.
+     */
+    fun initialize() = coroutineScope.launch {
+        val autoPushObserver = AutoPushObserver(accountManager, pushFeature, pushScope.value())
```

Comment:
nit: Let's assign `pushScope.value` to a `val` here? We're calling it twice right now, and it unconditionally generates a UUID and scope internally because you're using it as a default in `getString`.